File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int v1=0, v2=0;
        for(int i=0; i<nums.size(); i++){
            if(v1<=nums[i]){
                v2=v1;
                v1= nums[i];
            }
            else if(nums[i]<=v1 && v2<nums[i]) v2= nums[i];
        }
        v1--, v2--;
        return v1*v2;
    }
};

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.java
class Solution {
    public int maxProduct(int[] nums) {
        int max1=0, max2=0;
        for(int i=0; i<nums.length; i++){
            if(max1<nums[i]){
                max2= max1;
                max1= nums[i];
            }else if(max2<nums[i]) max2= nums[i];
        }
        return (max1-1)*(max2-1);
    }
}